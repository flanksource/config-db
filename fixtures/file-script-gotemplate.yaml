apiVersion: configs.flanksource.com/v1
kind: ScrapeConfig
metadata:
  name: file-script-gotemplate-scraper
spec:
  file:
    - type: MyConfig
      id: "$.id"
      name: "scraped"
      transform:
        script:
          gotemplate: |+
            [
              {
                {{range $i := .config}}
                  "name-{{.id}}": "hi {{.name}}",
                {{end}}
                "id": "1"
              }
            ]
      paths:
        - fixtures/data/multiple-configs.json
