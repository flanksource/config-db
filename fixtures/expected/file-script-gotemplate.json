[
  {
    "config_class": "MyConfig",
    "config_type": "MyConfig",
    "name": "scraped",
    "id": "1",
    "config": {
      "name-1": "hi Config1",
      "name-2": "hi Config2",
      "id": "1"
    }
  }
]
